Source/KakapoServer.swift

```diff
@@ -0,0 +1,92 @@
+//
+//  KakapoServer.swift
+//  KakapoExample
+//
+//  Created by Joan Romano on 31/03/16.
+//  Copyright © 2016 devlucky. All rights reserved.
+//
+
+import Foundation
+
+class KakapoServer: NSURLProtocol {
+    
+    typealias KakapoLookupObject = (method: KakapoServerMethod,
+                                    handler: (request: KakapoRequest) -> ())
+    
+    enum KakapoServerMethod: String {
+        case GET, POST, PUT, DELETE
+    }
+    
+    struct KakapoRequest {
+        let urlString: String
+        let params: [String : AnyObject]
+    }
+    
+    private static var routes: [String : KakapoLookupObject] = [:]
+    
+    class func enable() {
+        NSURLProtocol.registerClass(self)
+    }
+    
+    class func disable() {
+        NSURLProtocol.unregisterClass(self)
+    }
+    
+    override class func canInitWithRequest(request: NSURLRequest) -> Bool {
+        guard let requestString = request.URL?.absoluteString else { return false }
+        
+        for (key, object) in routes {
+            if object.method.rawValue == request.HTTPMethod && checkUrl(key, requestUrl: requestString) != nil {
+                return true
+            }
+        }
+        
+        return false
+    }
+    
+    override class func canonicalRequestForRequest(request: NSURLRequest) -> NSURLRequest {
+        return request
+    }
+    
+    override class func requestIsCacheEquivalent(a: NSURLRequest, toRequest b: NSURLRequest) -> Bool {
+        return false
+    }
+    
+    override func startLoading() {
+        guard let requestString = request.URL?.absoluteString else { return }
+        
+        for (key, object) in KakapoServer.routes {
+            if let params = checkUrl(key, requestUrl: requestString) {
+                object.handler(request: KakapoRequest(urlString: requestString, params: params))
+            }
+        }
+        
+        let response = NSHTTPURLResponse(URL: request.URL!, statusCode: 200, HTTPVersion: "", headerFields: nil)
```

Comment:
We must provide a flexible way of return `statusCode`. I'll suggest to let the user to return the status code from the `requestHandler`, since is the place in which we have all the context, think about the common case of making a request for an unexisting user or asking for a user which doesn't have permission to that enpoint. Is in the request handler where we know about that...

So I'll like to be able to do something like

```swift
let statusCode = object.handler(request: KakapoRequest(urlString: requestString, params: params))

let response = NSHTTPURLResponse(URL: request.URL!, statusCode: statusCode, HTTPVersion: "", headerFields: nil)
 
```